homeassistant/components/tplink/entity.py

```diff
@@ -311,33 +331,49 @@ def _category_for_feature(feature: Feature | None) -> EntityCategory | None:
 def _description_for_feature[_D: EntityDescription](
     desc_cls: type[_D],
     feature: Feature,
+    descriptions: Mapping[str, _D],
+    *,
+    child_alias: str | None = None,
 ) -> _D:
     """Return description object for the given feature.
 
     This is responsible for setting the common parameters & deciding based on feature id
     which additional parameters are passed.
     """
 
-    kwargs: dict = {}
-    # Key can be overridden for legacy support
-    if (entity_extras := ENTITY_EXTRAS.get(feature.id)) and (
-        key := entity_extras.get("key")
-    ):
-        kwargs["key"] = key
-    else:
-        kwargs["key"] = feature.id
-    # Use translation keys for all features configured in extras
-    if entity_extras:
-        kwargs["translation_key"] = feature.id
-        kwargs["state_class"] = entity_extras.get("state_class")
-        kwargs["device_class"] = entity_extras.get("device_class")
+    enabled_default = feature.category is not Feature.Category.Debug
+    category = _category_for_feature(feature)
+    if descriptions and (desc := descriptions.get(feature.id)):
+        # The state feature gets the device name or the child device
+        # name if it's a child device
+        if feature.id == PRIMARY_STATE_ID:
+            translation_key = None
+            entity_name: str | None | UndefinedType = child_alias
+        else:
+            translation_key = feature.id
+            entity_name = UNDEFINED
+        desc = replace(
+            desc,
+            translation_key=translation_key,
+            name=entity_name,
+            entity_category=category,
+            entity_registry_enabled_default=enabled_default,
+        )
     else:
-        kwargs["name"] = feature.name
-        kwargs["icon"] = feature.icon
-
-    kwargs["entity_registry_enabled_default"] = (
-        feature.category is not Feature.Category.Debug
-    )
-    kwargs["entity_category"] = _category_for_feature(feature)
-
-    return desc_cls(**kwargs)
+        # Entity are named in the base class based on the following logic:
+        # _attr_name > translation.name > description.name
+        # > device_class if base platform supports.
```

Comment:
The meaning of this is a bit unclear, is this our internal logic or the logic of homeassistant's way to create names?

Reply:
Done in https://github.com/home-assistant/core/pull/120064